### PR TITLE
[#4] Add list of predefined files to 'life init'

### LINKED ISSUE #4: Implement 'life init' command
- [ ] `life init <GitHub userName>`:
    * [x] Creates repository `~/dotfiles`, creates `~/.life` file
    * [x] Checks for errors
    * [ ] Provides some set of standard files (`.life`, `.profile`, `.vimrc`, `.ghc/`, `.spacemacs`,

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #4

This is how it looks like. I'm not satisfied with the output (alignment is slightly mixed) but dunno what to do with it... Any suggestions?

Init example

### CODE REVIEW

File: src/Life/Main/Init.hs [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
+predefinedLifeConfig = mempty
+    { lifeConfigurationFiles = Set.fromList
+          [ $(mkRelFile ".bash_profile")
+          , $(mkRelFile ".profile")
+          , $(mkRelFile ".vimrc")
+          , $(mkRelFile ".emacs")
+          , $(mkRelFile ".spacemacs")
+          , $(mkRelFile ".gitconfig")
+          , $(mkRelFile ".ghc/ghci.conf")
+          , $(mkRelFile ".stylish-haskell.yaml")
```
Comment (ChShersh) [inline src/Life/Main/Init.hs]: Please, feel free to give any suggestions what else can be added to this default list.

File: src/Life/Main/Init.hs [hunk 2]
```diff
@@ -35,8 +56,42 @@ lifeInit owner = do
...
+
+        unless (noExist == mempty) $ do
+            infoMessage "The following files and directories weren't found (so just ignoring them):"
+            skipMessage $ renderLifeConfiguration noExist
+
+        unless (exist == mempty) $ do
+            infoMessage "Found the following files and directories:"
+            successMessage $ renderLifeConfiguration exist
+
+        useDiscovered <- chooseYesNo "Would you like to add all discovered existing files and directories to .life configuration?"
```
Comment (ChShersh) [inline src/Life/Main/Init.hs]: This line is very long... Dunno what to do with long string messages.

File: src/Life/Main/Init.hs [hunk 3]
```diff
@@ -35,8 +56,42 @@ lifeInit owner = do
...
-    writeConf = do
-        infoMessage "Writing .life configuration file"
-        let lifeConfig = singleFileConfig lifePath
+    createLifeFile :: IO ()
+    createLifeFile = do
+        infoMessage "Checking existence of some commonly used predefined files..."
+        (exist, noExist) <- scanConfig predefinedLifeConfig
+
+        unless (noExist == mempty) $ do
+            infoMessage "The following files and directories weren't found (so just ignoring them):"
```
Comment (vrom911) [inline src/Life/Main/Init.hs]: (so just ignoring them) sounds a bit informal. Probably which will be ignored? Or even better, but I can't think of

File: src/Life/Main/Init.hs [hunk 4]
```diff
@@ -35,8 +56,42 @@ lifeInit owner = do
...
+
+        infoMessage "Initializing global .life configuration file..."
         writeGlobalLife lifeConfig
+
+{- | Split given configuration into two:
+
+1. All files and directories which exist on machine.
+2. Other non-existing files and dirs.
+-}
+scanConfig :: LifeConfiguration -> IO (LifeConfiguration, LifeConfiguration)
```
Comment (vrom911) [inline src/Life/Main/Init.hs]: That's really nice that you prepared the interface already for any further configs (not only list of predefined files) 👍

**Comment by ChShersh** [comment]
Now it looks like this:

screenshot from 2018-06-24 14-02-48

**Pull request merged**